src/app/sale/sale.model.ts

```diff
@@ -0,0 +1,56 @@
+import { Document, Schema } from 'mongoose';
+import { ICake } from '../cake/cake.interface';
+
+import { CakeModel } from '../cake/cake.model';
+import { ISale } from './sale.interface';
+
+export const SaleSchema = new Schema({
+    customerName: {
+        type: Schema.Types.String,
+        required: true,
+        minlength: 3,
+        maxlength: 50
+    },
+    customerPhoneNumber: {
+        type: Schema.Types.String,
+        required: true,
+        minlength: 10,
+        match: /^[+]*[(]{0,1}[0-9]{1,4}[)]{0,1}[-\s\./0-9]*$/
+    },
+    customerEmail: {
+        type: Schema.Types.String,
+        minlength: 5,
+        maxlength: 100,
+        match: /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/
+    },
+    totalAmount: {
+        type: Schema.Types.Number,
+        required: true,
+        min: 1
+    },
+    cakeId: {
+        type: Schema.Types.ObjectId,
+        ref: 'Cake',
+        required: true
+    }
+});
+
+SaleSchema.pre<ISale & Document>('save', async function(next) {
+    // const model: ICake = this.model('Cake');
```

Comment:
???

Reply:
This is where I have some issues since if I try to execute the commented code I will get an error about cake being null or Document and I didn't knew how to solve this issue

Maybe can you provide me a little guidance here? @nestorvanz

Reply:
Well, what I can tell you: This code should not be here, it's a better idea to move your business logic to the Class file, this will change your design centered on the domain (Domain-Drive Design)